src/app/utils/ServerApiClient.js

```diff
@@ -89,6 +89,12 @@ export function saveCords(x, y) {
     fetch('/api/v1/save_cords', request);
 }
 
+export function setUserPreferences(payload) {
+    if (!process.env.BROWSER || window.$STM_ServerBusy) return Promise.resolve();
```

Comment:
does this not mean this function silently short-circuits if `window.$STM_ServerBusy` is set? Shouldn't we reject immediately in that case, or wait around until the server's not busy anymore?

Reply:
Server Busy usually means that mysql is not available so there is no point in saving user preferences at this point.
window.$STM_ServerBusy can only change on page reload.